packages/Search/test/tstLinearBVH.cpp

```diff
@@ -41,134 +41,289 @@ void check_results( DataTransferKit::BVH<DeviceType> &bvh,
     TEST_COMPARE_ARRAYS( offset_host, offset_ref );
 }
 
-namespace details = DataTransferKit::Details;
+// Same as above except that we get the distances out of the queries and
+// compare them to the reference solution passed as argument.  Templated type
+// `Query` is pretty much a nearest predicate in this case.
+template <typename Query, typename DeviceType>
+void check_results( DataTransferKit::BVH<DeviceType> &bvh,
+                    Kokkos::View<Query *, DeviceType> const &queries,
+                    std::vector<int> const &indices_ref,
+                    std::vector<int> const &offset_ref,
+                    std::vector<double> const &distances_ref, bool &success,
+                    Teuchos::FancyOStream &out )
+{
+    Kokkos::View<int *, DeviceType> indices( "indices" );
+    Kokkos::View<int *, DeviceType> offset( "offset" );
+    Kokkos::View<double *, DeviceType> distances( "distances" );
+    bvh.query( queries, indices, offset, distances );
 
-template <typename DeviceType>
-class FillBoxes
+    auto indices_host = Kokkos::create_mirror_view( indices );
+    deep_copy( indices_host, indices );
+    auto offset_host = Kokkos::create_mirror_view( offset );
+    deep_copy( offset_host, offset );
+    auto distances_host = Kokkos::create_mirror_view( distances );
+    deep_copy( distances_host, distances );
+
+    TEST_COMPARE_ARRAYS( indices_host, indices_ref );
+    TEST_COMPARE_ARRAYS( offset_host, offset_ref );
+    TEST_COMPARE_FLOATING_ARRAYS( distances_host, distances_ref, 1e-14 );
+}
+
+// Hopefully we can git rid of this once the operator== has been implemented
+// for boxes.
+void test_box_equality( DataTransferKit::Box const &l,
+                        DataTransferKit::Box const &r, bool &success,
+                        Teuchos::FancyOStream &out )
 {
-  public:
-    KOKKOS_INLINE_FUNCTION
-    FillBoxes( Kokkos::View<DataTransferKit::Box *, DeviceType> boxes )
-        : _boxes( boxes )
-    {
-    }
+    TEST_EQUALITY( l[0], r[0] );
+    TEST_EQUALITY( l[1], r[1] );
+    TEST_EQUALITY( l[2], r[2] );
+    TEST_EQUALITY( l[3], r[3] );
+    TEST_EQUALITY( l[4], r[4] );
+    TEST_EQUALITY( l[5], r[5] );
+}
 
-    KOKKOS_INLINE_FUNCTION
-    void operator()( int const i ) const
-    {
-        if ( i == 0 )
-            _boxes[0] = {{0, 0, 0, 0, 0, 0}};
-        else
-            _boxes[1] = {{1, 1, 1, 1, 1, 1}};
-    }
+template <typename DeviceType>
+DataTransferKit::BVH<DeviceType>
+make_bvh( std::vector<DataTransferKit::Box> const &b )
```

Comment:
Do we want to move all these move all these `make_` functions outside of test?

Reply:
Move them out of this test into some header for use in testing for the Search package? Maybe (only if it is actually needed elsewhere).
Make them more broadly accessible? No, these are cheap convenience helpers, they would kill performance and we certainly don't want the user to believe that's the proper way to build them.